controllers/vaultauth_controller.go

```diff
@@ -64,13 +68,48 @@ func (r *VaultAuthReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 	if o.GetDeletionTimestamp() != nil {
 		logger.Info("Got deletion timestamp", "obj", o)
+		r.referenceCache.Remove(VaultAuthGlobal, req.NamespacedName)
 		return r.handleFinalizer(ctx, o)
 	}
 
 	// assume that status is always invalid
 	o.Status.Valid = false
-
 	var errs error
+
+	var conditions []metav1.Condition
+	if o.Spec.VaultAuthGlobalRef != "" {
+		globalRef, err := common.ParseResourceRef(o.Spec.VaultAuthGlobalRef, o.GetNamespace())
+		r.referenceCache.Set(
+			VaultAuthGlobal, req.NamespacedName,
+			globalRef)
+
+		condition := metav1.Condition{
+			Type:               "VaultAuthGlobalRef",
+			Status:             "True",
+			ObservedGeneration: o.Generation,
+			LastTransitionTime: metav1.NewTime(nowFunc()),
+			Reason:             "VaultAuthGlobalRef",
```

Comment:
I wonder if `Type` and `Reason` be different?

Reply:
https://github.com/hashicorp/vault-secrets-operator/pull/850